packages/cozy-harvest/src/Manifest.js

```diff
@@ -0,0 +1,63 @@
+export const legacyLoginFields = [
+  'login',
+  'identifier',
+  'new_identifier',
+  'email'
+]
+
+export const roleShouldBeRequired = ['identifier', 'password']
+
+export const legacyEncryptedFields = [
+  'secret',
+  'dob',
+  'code',
+  'answer',
+  'access_token',
+  'refresh_token',
+  'appSecret'
+]
+
+export const Manifest = {
+  sanitize: manifest => {
+    if (!manifest.fields) return manifest
+    manifest = Manifest.sanitizeIdentifier(manifest)
+    manifest = Manifest.sanitizeIsRequired(manifest)
+    manifest = Manifest.sanitizeEncrypted(manifest)
+    return manifest
+  },
+
+  sanitizeIdentifier: manifest => {
+    for (let fieldName in manifest.fields)
+      if (manifest.fields[fieldName].role === 'identifier') return manifest
```

Comment:
I think it does not clean the case when several fields are marked as `identifier`.